### DISCUSSION TITLE
Is there a way to use sbt bsp instead of bloop by default?

### DISCUSSION THREAD

**Discussion by antosha417** [scalameta/metals#5658] [Q&A] [answered]
I feel that my experience with sbt bsp is better than with bloop and was wandering why do I have to switch build server in every project I open. Sure there should be some setting in nvim-metals I'm not aware of to use sbt instead of bloop by default

**Comment by ckipp01** [comment, marked as answer]
Hey @antosha417, good question. This is something that we've discussed and that we might pursue, but as of now there is no way to set it as a default.

**Reply by tkolleh** [reply]
Is there a way to force this as the default for a repo today. I'm building Java IDL/thrift files for a legacy project and need to use sbt BSP everytime.

https://github.com/scalameta/metals/issues/5280#issuecomment-1566715271

**Reply by tgodzik** [reply]
There is metals.defaultBspToBuildTool option that should work

**Reply by linux-root** [reply]
@tgodzik is there any option to change memory of BSP server? Bloop is quite problematic in my case so I decided to use sbt BSP instead, however it seems  that 100m memory is too small for my project. Logs from metals:
```shell
2025.12.17 15:11:45 INFO  Running BSP server List(/Users/root/.jdks/zulu17.58.21-ca-jdk17.0.15-macosx_aarch64/zulu-17.jdk/Contents/Home/bin/java, -Xms100m, -Xmx100m, -classpath, /Users/root/Library/Caches/Coursi
e
```

**Reply by linux-root** [reply]
@ckipp01 do you think that the option `metals.defaultBspToBuildTool ` can be used in `metals-nvim` ? I tried this but it seems not working:
```lua
    opts = function()
      local metals_config = require("metals").bare_config()
      metals_config.settings = {
        defaultBspToBuildTool = true,
      }
      return metals_config
    end,
```

**Reply by tgodzik** [reply]
You should be able to change the memory with .sbtopts file. It should be taken into account when starting the BSP server

**Reply by linux-root** [reply]
> You should be able to change the memory with .sbtopts file. It should be taken into account when starting the BSP server

I did have .sbtopts but it seemed not working. I can manually change jvm opt of the bsp server by manually editting the file `.bsp/sbt.json` and see it works:
```json
{"name":"sbt","version":"1.11.7","bspVersion":"2.1.0-M1","languages":["scala"],"argv":["/Users/root/.jdks/zulu17.58.21-ca-jdk17.0.15-macosx_aarch64/zulu-17.jdk/Contents/Home/bin/java","-Xms100m","-Xmx100m",
"-classpath","/Users/root/Library/Caches/Coursier/arc/https/github.com/sbt/sbt/releases/download/v1.11.4/sbt-1.11.4.zip/sbt/bin/sbt-launch.jar",
"-Dsbt.script=/Users/root/Library/Caches/Coursier/arc/https/github.com/sbt/sbt/releases/download/v1.11.4/sbt-1.11.4.zip/sbt/bin/sbt","xsbt.boot.Boot","-bsp"]}
```
However `.bsp/sbt.json` is generated by Metals, so I guess it'd be better to set those values (in `argv`)  via Metal. Do we have any option like that @tgodzik  ?

**Reply by tgodzik** [reply]
Ok, looks like what you see in sbt.json that is jsut for the launcher. But .jvmopts file seems to be picked up by sbt

**Reply by tgodzik** [reply]
You will need to restart sbt though.

**Reply by linux-root** [reply]
I see. Thank you!
